streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordDeserializer.java

```diff
@@ -80,12 +81,29 @@ public static void handleDeserializationFailure(final DeserializationExceptionHa
                                                     final ConsumerRecord<byte[], byte[]> rawRecord,
                                                     final Logger log,
                                                     final Sensor droppedRecordsSensor) {
+        handleDeserializationFailure(deserializationExceptionHandler, processorContext, deserializationException, rawRecord, log, droppedRecordsSensor, null);
```

Comment:
Why do we still need this one? Can't we update `GlobalStateManagerImpl` to also pass in a `sourceNode.name` ?

Reply:
@sebastienviale What about this comments? Merge the PR for now to make progress.

Reply:
@mjsax I've been looking for a way to update `GlobalStateManagerImpl` so we can remove this method. But it looks like we have no source node name in `GlobalStateManagerImpl`. There's a "source" processor: 

https://github.com/apache/kafka/blob/010ab19b724ae011e85686ce47320f4f85d9a11f/streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java#L315

But it doesn't look to be named. E.g., in the following scenario:

```java
streamsBuilder
    .addGlobalStore(storeBuilder, "PERSON_TOPIC",
        Consumed.with(Serdes.String(), SerdesUtils.specificAvroValueSerdes()),
        () -> (Processor<String, SpecificRecord, Void, Void>) record -> log.info("Processing record with key = {}, value = {}", record.key(), record.value())); 
// Source var is this processor ⬆️
```

Reply:
Well, the "helper" method also only calls the second one, setting `sourceNodeName` to `null`. Why can't we just call the second one from `GlobalStateManagerImpl` directly also setting it to `null`, and thus avoiding the need for the overload? The overload does not really add much value, or does it?

Reply:
@mjsax Addressing it in https://github.com/apache/kafka/pull/16746